packages/hoppscotch-common/src/helpers/import-export/import/openapi.ts

```diff
@@ -627,14 +627,27 @@ const convertPathToHoppReqs = (
           ? openAPIUrl + openAPIPath.slice(1)
           : openAPIUrl + openAPIPath
 
+      let defaultRequestName = "Untitled Request"
+
+      if (info.tags?.length) {
+        const pathStartIndex =
```

Comment:
here this logic will miss some edge cases,

i belive you're assuming that the tag will always be present in the path, but that won't always be the case.

1. what happens if the tag is no way related to the path ?
2. adding the tag length makes sense only if the tag is found in the path, otherwise you're adding the length to -1, which does not makes much sense.

Reply:
No. If it does not have a tag then the condition is not met and is saved as a request without a title. What I can do is save it with the name of the controller then the request in case I don't have tags.

Reply:
I quoted the wrong lines actually. the comment is meant for the next line. 
```typescript
const pathStartIndex =
          openAPIPath.indexOf(info.tags[0]) + info.tags[0].length
```

and i didn't mean when the tag is not present, i meant when a tag is present but it is not in the url. 

like this,

```
/categories/{categoryId}/items:
    get:
      tags:
        - SomethingNonRelatedToTheURL
```